lib/posix-process/arch/x86_64/execve.c

```diff
@@ -0,0 +1,29 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <uk/arch/ctx.h>
+#include <uk/essentials.h>
+
+void execve_arch_execenv_init(struct ukarch_execenv *execenv,
+			      __uptr ip, __uptr sp)
+{
+	execenv->regs.rip = ip;
+	execenv->regs.rsp = sp;
+
+	/* Prepare for iret */
+	execenv->regs.eflags = X86_EFLAGS_IF; /* enable IRQs on ctx switch */
+	execenv->regs.cs = 8;
```

Comment:
`GDT_DESC_OFFSET(GDT_DESC_CODE)` and `GDT_DESC_OFFSET(GDT_DESC_DATA)`. yes this introduces deps to plat common :zany_face:

Reply:
I'm not sure we want to introduce such a dependency. @skuenzer?

Reply:
Can you also add a comment like a TODO or smth so that someone in the future may do this replacement once we get rid of this `plat/common` stuff?